SurfaceTopography/Uniform/PowerSpectrum.py

```diff
@@ -209,6 +209,133 @@ def power_spectrum_from_area(self, window=None, reliable=True, collocation='log'
         return q_val, C_val
 
 
+def fftfreq(t, meshgrid=True):
+    """
+    computes and returns the grid of wavevectors corresponding to np.fft.fft
+
+    """
+
+    # if t.dim == 1:
+    #     nb_grid_pts = [t.nb_grid_pts]
+    #     physical_sizes = [t.physical_sizes]
+
+    qx = 2 * np.pi * np.fft.fftfreq(
+        t.nb_grid_pts[0],
+        t.physical_sizes[0] / t.nb_grid_pts[0])
+    if t.dim == 1:
+        return qx
+    elif t.dim == 2:
+        qy = 2 * np.pi * np.fft.fftfreq(
+            t.nb_grid_pts[1],
+            t.physical_sizes[1] / t.nb_grid_pts[1])
+
+        if meshgrid:
+            qx, qy = np.meshgrid(qx, qy, indexing='ij')
+        return qx, qy
+
+
+def wavevectors_norm2(t, meshgrid=True):
+    if t.dim == 1:
+        return np.abs(t.fftfreq(meshgrid=meshgrid))
+    else:
+        qx, qy = t.fftfreq(meshgrid=meshgrid)
+        return np.sqrt(qx ** 2 + qy ** 2)
+
+
+UniformTopographyInterface.register_function("fftfreq", fftfreq)
+UniformTopographyInterface.register_function("wavevectors_norm2", wavevectors_norm2)
+
+
+def integrate_psd(self, factor=lambda q: 1, window=None, reliable=True, ):
+    """
+
+    factor: function
+        Function taking as argument the 2 norm of the wavevector (if it accepts only one argument)
+
+        or the two components of the wavevector qx and qy
+
+        # TODO: In future we might want to allow for giving
+    """
+    if self.has_undefined_data:
+        raise UndefinedDataError('This topography has undefined data (missing data points). Power-spectrum cannot be '
+                                 'computed for topographies with missing data points.')
+
+    h = self.window(window).heights()
+
+    fourier_topography = np.prod(self.physical_sizes) / np.prod(self.nb_grid_pts) * np.fft.fftn(h)
```

Comment:
This should probably run through `muFFT`

Reply:
I just took over what we did in previous code. We would need to change that at several other places.

Reply:
I opened an issue for changing that in the whole code